pkg/validation/node_conditions.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"github.com/golang/glog"
+	"k8s.io/api/core/v1"
+)
+
+func getNodeReadyStatus(node *v1.Node) v1.ConditionStatus {
+	cond := findNodeCondition(node, v1.NodeReady)
+	if cond != nil {
+		return cond.Status
+	}
+	return v1.ConditionUnknown
+}
+
+func findNodeCondition(node *v1.Node, conditionType v1.NodeConditionType) *v1.NodeCondition {
+	for i := range node.Status.Conditions {
+		cond := &node.Status.Conditions[i]
+		if cond.Type == conditionType {
+			return cond
+		}
+	}
+	return nil
+}
+
+// isNodeReady returns if a Node is considered ready.
+// It is considered ready if:
+// 1) its Ready condition is set to true
+// 2) doesn't have NetworkUnavailable condition set to true
+func isNodeReady(node *v1.Node) bool {
```

Comment:
So a lot of this code initially was taken from upstream from the e2e testing. Did you do the same, or did you do a general rewrite?

Reply:
The code that's left is the functions that were used, and then I collapsed the functions a little bit to be more readable.  Logic should be unchanged.